guidedmodules/views.py

```diff
@@ -426,6 +428,113 @@ def redirect_to():
         }
     )
 
+    # Process any actions from the question.
+    # --------------------------------------
+    # Can we automatically make other changes to the project, including the system
+    # based on an question answer?
+    # For example, and we add a component and its statements to a system
+    # based on what the user selects in the questions?
+    # This block processes any actions specified in the question.
+    # This requires a tightly controlled vocabulary.
+    #
+    # We assume user has sufficient permission because user is answering question.
+    #
+    if 'actions' in q.spec:
+        # Loop through list of actions
+        for action in q.spec['actions']:
+            # Perform action if question (task) `value` is same as defined action value
+            if value == action['value']:
+                # Get project_id to be compatible with borrowed code block
+                project_id = task.project_id
+                project = Project.objects.get(pk=project_id)
+                system = project.system
+                system_id = system.id
+                # Decode the action by splitting on `/`
+                a_obj, a_verb, a_filter = action['action'].split("/")
+
+                # Process element actions
+                # -----------------------------------
+                # Only two actions are currently supported:
+                #   1. `element/add_role/<role_value>` - Automatically add elements to the selected components of a system
+                #   2. `element/del_role/<role_value>` - Automatically delete elements from the selected components of a system
+                if a_obj == 'element':
+
+                    # Get all elements assigned role specified in the action
+                    elements_with_role = Element.objects.filter(roles__role=a_filter)
```

Comment:
```suggestion
                    elements_with_role = Element.objects.filter(element_type="system_element").filter(roles__role=a_filter)
```
Only want system_element not systems